src/xb-builder-source-ctx.c

```diff
@@ -112,9 +115,25 @@ xb_builder_source_ctx_get_content_type (XbBuilderSourceCtx *self,
 				      cancellable, error))
 			return NULL;
 		if (bufsz > 0)
-			return g_content_type_guess (priv->filename, buf, bufsz, NULL);
+			content_type = g_content_type_guess (priv->filename, buf, bufsz, NULL);
 	}
-	return g_content_type_guess (priv->filename, NULL, 0, NULL);
+
+	/* either unseekable, or empty */
+	if (content_type == NULL)
+		content_type = g_content_type_guess (priv->filename, NULL, 0, NULL);
+
+#ifdef __MINGW64__
```

Comment:
Should this maybe be a `#ifdef WIN32` instead?  So if someone decides to compile this with visual studio instead of mingw64 some day it still works?

Reply:
Is WIN32 defined for a 64 bit build?

Reply:
Looks like no; should be `_WIN64`
https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=vs-2019

Reply:
> Is WIN32 defined for a 64 bit build?

The answer to that one is yes, and it's defined in MinGW -- so I've switched to using that.